tests/tests_mqtt_server/src/server.rs

```diff
@@ -0,0 +1,66 @@
+use once_cell::sync::OnceCell;
+use std::{
+    net::TcpStream,
+    process::{Child, Command, Stdio},
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Mutex,
+    },
+    thread::sleep,
+    time::Duration,
+};
+
+static SERVER: OnceCell<Mutex<Child>> = OnceCell::new();
+static CALLERS_COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+fn start_server(port: u16) {
+    let _server_child_handle = SERVER.get_or_init(|| Mutex::new(spawn_server_process(port)));
+
+    CALLERS_COUNTER.fetch_add(1, Ordering::Relaxed);
+}
+
+fn try_stop_server() {
+    CALLERS_COUNTER.fetch_sub(1, Ordering::Relaxed);
+
+    if CALLERS_COUNTER.load(Ordering::Relaxed) == 0 {
```

Comment:
Shouldn't we loop until the count becomes 0? Otherwise we stand the risk of leaving the server running stale even after the test is complete and all references to this server are dropped?